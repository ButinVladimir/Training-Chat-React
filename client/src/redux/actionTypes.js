export const CHANGE_LOGIN = 'CHANGE_LOGIN';
export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';
export const CHANGE_SPEECH = 'CHANGE_SPEECH';
export const SAY = 'SAY';
export const ADD_TO = 'ADD_TO';
export const REMOVE_TO = 'REMOVE_TO';
export const CONNECT = 'CONNECT';
export const DISCONNECT = 'DISCONNECT';
export const SHOW_ERROR = 'SHOW_ERROR';
export const HIDE_ERROR = 'HIDE_ERROR';
export const USER_APPEARS = 'USER_APPEARS';
export const USER_DISAPPEARS = 'USER_DISAPPEARS';
export const GET_MESSAGE = 'GET_MESSAGE';
export const UPDATE_USER_LIST = 'UPDATE_USER_LIST';
