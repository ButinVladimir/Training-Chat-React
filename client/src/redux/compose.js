export default function compose(...args) {
  return (state, action) => {
    let resultState = state;
    args.forEach((reducer) => { resultState = reducer(resultState, action); });

    return resultState;
  };
}
