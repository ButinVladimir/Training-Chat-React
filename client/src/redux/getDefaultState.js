export default () => ({
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
  connected: false,
  errorPanel: {
    showError: false,
    error: '',
  },
});
