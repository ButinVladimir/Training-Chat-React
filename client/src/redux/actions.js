import * as actionTypes from './actionTypes';

export const changeLogin = loginValue => ({
  type: actionTypes.CHANGE_LOGIN,
  loginValue,
});

export const login = () => ({
  type: actionTypes.LOGIN,
});

export const logout = () => ({
  type: actionTypes.LOGOUT,
});

export const changeSpeech = speechValue => ({
  type: actionTypes.CHANGE_SPEECH,
  speechValue,
});

export const say = (id, date) => ({
  type: actionTypes.SAY,
  id,
  date,
});

export const addTo = recipient => ({
  type: actionTypes.ADD_TO,
  recipient,
});

export const removeTo = recipient => ({
  type: actionTypes.REMOVE_TO,
  recipient,
});
