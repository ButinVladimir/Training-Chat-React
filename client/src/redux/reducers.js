import compose from './compose';
import loginPanel from './reducers/loginPanel';
import speechPanel from './reducers/speechPanel';

const reducers = compose(loginPanel, speechPanel);

export default (state = {
  loginPanel: {
    loginned: false,
    login: '',
  },
  usersPanel: {
    usersList: [],
  },
  speechPanel: {
    speech: '',
    to: [],
  },
  chatPanel: {
    messages: [],
  },
}, action) => reducers(state, action);
