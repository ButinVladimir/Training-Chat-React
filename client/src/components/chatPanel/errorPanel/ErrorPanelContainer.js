import { connect } from 'react-redux';
import ErrorPanel from './ErrorPanel';
import { hideError } from '../../../redux/actions';

const mapStateToProps = state => ({
  error: state.errorPanel.error,
});

const mapDispatchToProps = dispatch => ({
  onCloseError: () => dispatch(hideError()),
});

const ErrorPanelContainer = connect(mapStateToProps, mapDispatchToProps)(ErrorPanel);

export default ErrorPanelContainer;
