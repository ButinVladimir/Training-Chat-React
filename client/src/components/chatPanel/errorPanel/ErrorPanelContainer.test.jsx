import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import ErrorPanelContainer from './ErrorPanelContainer';
import { hideError } from '../../../redux/actions';

describe('ErrorPanelContainer', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore([]);
  });

  it('renders error panel', () => {
    const store = mockStore({
      errorPanel: {
        showError: true,
        error: 'error',
      },
    });
    const component = mount(
      <Provider store={store}>
        <ErrorPanelContainer />
      </Provider>,
    );

    expect(component).toMatchSnapshot();
  });

  it('handles closing error panel', () => {
    const store = mockStore({
      errorPanel: {
        showError: true,
        error: 'error',
      },
    });
    const component = mount(
      <Provider store={store}>
        <ErrorPanelContainer />
      </Provider>,
    );

    let actions = store.getActions();
    let expectedActions = [];
    expect(actions).toEqual(expectedActions);

    component.find('.close-button').simulate('click');

    actions = store.getActions();
    expectedActions = [hideError()];
    expect(actions).toEqual(expectedActions);
  });
});
