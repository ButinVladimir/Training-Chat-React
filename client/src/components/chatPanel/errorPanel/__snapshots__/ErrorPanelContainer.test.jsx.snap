// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPanelContainer renders error panel 1`] = `
<div
  className="error-panel-container"
>
  <div
    className="error-panel"
  >
    <div
      className="popup-row"
    >
      An error occurred
    </div>
    <div
      className="popup-row"
    >
      error
    </div>
    <div
      className="popup-row"
    >
      <button
        className="close-button"
        onClick={[Function]}
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
